x-pack/plugins/ingest_pipelines/public/url_generator.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'src/core/public';
+import { UrlGeneratorsDefinition } from '../../../../src/plugins/share/public';
+import { INGEST_PIPELINES_PAGES, URL_GENERATOR } from './application/services/navigation';
+import { Dependencies } from './types';
+import { MANAGEMENT_APP_ID, PLUGIN_ID } from '../common/constants';
+
+export const INGEST_PIPELINES_APP_ULR_GENERATOR = 'INGEST_PIPELINES_APP_URL_GENERATOR';
+
+interface UrlGeneratorState {
+  pipelineId: string;
+  absolute?: boolean;
+}
+export interface PipelinesListUrlGeneratorState extends Partial<UrlGeneratorState> {
+  page: INGEST_PIPELINES_PAGES.LIST;
+}
+
+export interface PipelineEditUrlGeneratorState extends UrlGeneratorState {
+  page: INGEST_PIPELINES_PAGES.EDIT;
+}
+
+export interface PipelineCloneUrlGeneratorState extends UrlGeneratorState {
+  page: INGEST_PIPELINES_PAGES.CLONE;
+}
+
+export interface PipelineCreateUrlGeneratorState extends UrlGeneratorState {
+  page: INGEST_PIPELINES_PAGES.CREATE;
+}
+
+export type IngestPipelinesUrlGeneratorState =
+  | PipelinesListUrlGeneratorState
+  | PipelineEditUrlGeneratorState
+  | PipelineCloneUrlGeneratorState
+  | PipelineCreateUrlGeneratorState;
+
+export class IngestPipelinesUrlGenerator
+  implements UrlGeneratorsDefinition<typeof INGEST_PIPELINES_APP_ULR_GENERATOR> {
+  constructor(private readonly getAppBasePath: (absolute: boolean) => Promise<string>) {}
+
+  public readonly id = INGEST_PIPELINES_APP_ULR_GENERATOR;
+
+  public readonly createUrl = async (state: IngestPipelinesUrlGeneratorState): Promise<string> => {
+    return `${await this.getAppBasePath(!!state.absolute)}${URL_GENERATOR[state.page](
+      state.pipelineId!
```

Comment:
We're referring to this as `pipelineId` here but in `navigation.ts` the term `name` is used instead. Can we consistently use one or the other to make the connection clearer?

Reply:
Hey @cjcenizal , I'm glad you noticed that! After a discussion with @darnautov and reading the docs, I figured out that the "official" term is `pipeline_id`. In Ingest Pipelines plugin code we refer to this property as `pipelineName`, so I decided to use `name` in internal functions and `id` for external use.